Enable parallel clang-tidy on ec2 runner

1ntEgr8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60871 [test] Run parallel clang-tidy on codebase (failure expected)
* **#60870 Enable parallel clang-tidy on ec2 runner**
* #60869 Print stdout and stderr to console on parallel runs

This PR makes `clang-tidy` run on our self-hosted runner in a parallel fashion.

Fixes #60867
Test Plan: #60871

Differential Revision: [D29434240](https://our.internmc.facebook.com/intern/diff/D29434240)

1ntEgr8 committed:
Enable parallel clang-tidy on ec2 runner

[ghstack-poisoned]

driazati commented on .github/workflows/lint.yml:
```diff
@@ -329,6 +329,7 @@ jobs:
           # deploy/interpreter files are excluded due to using macros and other techniquies
           # that are not easily converted to accepted c++
           python3 tools/linter/clang_tidy.py \
+            -j \
```
nit: use full flag in scripts

```suggestion
            --parallel \
```

1ntEgr8 committed:
Update .github/workflows/lint.yml

Co-authored-by: driazati <9407960+driazati@users.noreply.github.com>

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29434240).

1ntEgr8 committed:
Update on "Enable parallel clang-tidy on ec2 runner"


This PR makes `clang-tidy` run on our self-hosted runner in a parallel fashion.

Fixes #60867
Test Plan: #60871

Differential Revision: [D29434240](https://our.internmc.facebook.com/intern/diff/D29434240)

[ghstack-poisoned]

1ntEgr8 commented:
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29434240).
